lib/ansible/modules/cloud/amazon/ec2_vol.py

```diff
@@ -379,9 +411,62 @@ def create_volume(module, ec2, zone):
                 ec2.create_tags([volume.id], tags)
         except boto.exception.BotoServerError as e:
             module.fail_json(msg = "%s: %s" % (e.error_code, e.error_message))
+    else:
+        changed = modify_volume(module, ec2, volume)
 
     return volume, changed
 
+def get_boto3_ec2_connection(module):
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        if not region:
+            module.fail_json(msg="region must be specified")
+        return boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except boto.exception.NoAuthHandlerFound as e:
+        module.fail_json(msg="Can't authorize connection - "+str(e))
```

Comment:
You could use traceback here. You'll need to import it too. https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#exception-handling